Cymptom

chenerlich opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Add Cymptom as a partner to Demisto.

## Screenshots
![image](https://user-images.githubusercontent.com/17673075/99910543-4b7d9e00-2cf7-11eb-9bce-ebac86414b7c.png)
![image](https://user-images.githubusercontent.com/17673075/99910562-694b0300-2cf7-11eb-9cd5-f962a14d2e9d.png)


## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=10091) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=10091) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=10091) it.</sub>

Bargenish commented on Packs/Cymptom/Playbooks/Cymptom_-_Expire_passwords_for_password_cracked_users.yml:
```diff
@@ -0,0 +1,117 @@
+description: Expires passwords for password cracked users using an Active Directory
```
please add a detailed use-case and description for this playbook.

Bargenish reviewed (commented):
@chenerlich Thank you for your contribution! great work.
Please review my comment.

Bargenish commented:
Hi there @chenerlich  if you are aiming to contribute this pack please address the relevant comments so we can get this pack merged, if not please let us know as soon as possible, as we are allocating resources to handle this contribution process.

chenerlich commented:
Hi @Bargenish, sorry for the delay and thanks for the PR!
I addressed some of the issues already, but didn't finish yet.
I'll push during this week when It's finished.
Thanks!

chenerlich commented on Packs/Cymptom/Integrations/Cymptom/Cymptom.yml:
```diff
@@ -0,0 +1,106 @@
+  the existing approach by transforming attack simulation into a data analysis question.
+  Cymptom agentless scanning brings real-time always-on visibility into the entire
+  security posture.
+display: Cymptom (Partner Contribution) (Partner Contribution) (Partner Contribution)
+  (Partner Contribution) (Partner Contribution) (Partner Contribution) (Partner Contribution)
```
I think it's a bug on the platform, since I didn't touch it..

chenerlich commented on Packs/Cymptom/Integrations/Cymptom/Cymptom.py:
```diff
@@ -0,0 +1,284 @@
+        """
+        results = self._http_request(
+            method='GET',
+            url_suffix='test'
```
Indeed

chenerlich commented on Packs/Cymptom/Integrations/Cymptom/Cymptom.py:
```diff
@@ -0,0 +1,284 @@
+        return return_error("There was an error")
+
+
+def fetch_incidents(client: Client) -> None:
```
I'll remove it for now.

Bargenish replied:
let me know if you need any assistance  with implementing the `fetch-incident` command.

chenerlich replied:
Thanks. 
I'll address it in the next version since we have some changes to process in backend before that.

chenerlich commented on Packs/Cymptom/Integrations/Cymptom/README.md:
````diff
@@ -0,0 +1,92 @@
+``` 
+!cymptom-get-mitigations
+```
+
````
Added initially but it's too much info with too many details to censor which makes it pretty unreadable

chenerlich commented on Packs/Cymptom/Integrations/Cymptom/Cymptom.py:
```diff
@@ -0,0 +1,284 @@
+    state = args.get("state", MitigationsState.open.value)
+    timeout = int(timeout)
+
+    mitigations_results = client.get_mitigations(timeout=timeout, mitigations_state=state)
```
It's not much and backend doesn't support limit (tho added as a task).
Thanks

chenerlich commented:
@Bargenish - Hi Bar,
Updated PR. 
Please let me know if there something else I need to do (:

Bargenish commented on Packs/Cymptom/Integrations/Cymptom/Cymptom.py:
```diff
@@ -0,0 +1,250 @@
+                                      "SubTechniques": extended_info["subtechniques"],
+                                      "References": extended_info["references"]})
+
+    readable_output = tableToMarkdown('Mitigations', mitigations_formatted, headers=table_headers)
```
If you want the headers to be human-readable you can pass an headerTransform function.
```suggestion
    readable_output = tableToMarkdown('Mitigations', mitigations_formatted, headers=table_headers, headerTransform=pascalToSpace)
```

chenerlich replied:
I'll keep it for next PRs (:

Bargenish reviewed (commented):
@chenerlich great work on the changes.
See my comments.

Let's also schedule a demo.

Bargenish commented on Packs/Cymptom/Integrations/Cymptom/Cymptom.py:
```diff
@@ -0,0 +1,250 @@
+        "Authorization": f"Bearer {api_key}"  # Replace ${token} with the token you have obtained
+    }
+
+    demisto.debug(" ---- MAIN CALL -----")
+    demisto.debug(" ---- PARAMS -----")
+    demisto.debug(f"base_url: {base_url}")
+    demisto.debug(f"api_key: {api_key}")
+    demisto.debug(f"insecure: {insecure}")
+    demisto.debug(f"use_proxy: {use_proxy}")
```
Also, let's remove these. You can find all the integration's parameters in the log bundle.
If you still like to keep those, please remove the api_key log. it is not secured and this field is encrypted.

chenerlich replied:
Isn't the api key controlled only by the user aa the user is the one who filled it?
Also, where can I find the log bundle?

chenerlich replied:
Isn't the api key controlled only by the user aa the user is the one who filled it?
Also, where can I find the log bundle?

Bargenish replied:
I dont fully understand what you mean by saying that the API key is controlled by the user.
API key is not something that we want to log. it creates a security hole.

You can read about log bundle [here](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-0/cortex-xsoar-admin/logs/create-a-log-bundle.html)
and read about executing a command with `debug-mode=true` [here](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-0/cortex-xsoar-admin/logs/create-a-log-bundle.html)

chenerlich replied:
Gotcha.

chenerlich commented on Packs/Cymptom/Integrations/Cymptom/README.md:
```diff
@@ -0,0 +1,92 @@
+Cymptom is a Breach and Attack Simulation solution that revolutionizes the existing approach by transforming attack simulation into a data analysis question. Cymptom agentless scanning brings real-time always-on visibility into the entire security posture.
+This integration was integrated and tested with version xx of Cymptom
```
![image](https://user-images.githubusercontent.com/17673075/103570306-89163f00-4ed1-11eb-9de4-53b316fa2587.png)
I actually have. Weird.

chenerlich replied:
It's also updated in the forked repo:
![image](https://user-images.githubusercontent.com/17673075/103630173-f9f93d80-4f49-11eb-9215-371f290db70c.png)

Bargenish commented:
@chenerlich sent you a mail to schedule a demo.

Bargenish commented:
@chenerlich any updates here?

chenerlich commented:
@Bargenish I've decreased the output size by removing a few headers. Let me know if it's OK now. Can send you a screen shot on slack as well.

Bargenish commented:
@chenerlich Please attach a screenshot of the outputs after executing the command.

chenerlich commented:
@Bargenish 
![image](https://user-images.githubusercontent.com/17673075/106434653-84b25700-647a-11eb-85fd-f90d21996e1f.png)
![image](https://user-images.githubusercontent.com/17673075/106439467-651e2d00-6480-11eb-9a66-b206dea944d1.png)
